src/TwigComponent/src/Twig/ComponentExtension.php

```diff
@@ -53,11 +54,27 @@ public function getTokenParsers(): array
 
     public function render(string $name, array $props = []): string
     {
-        return $this->container->get(ComponentRenderer::class)->createAndRender($name, $props);
+        try {
+            return $this->container->get(ComponentRenderer::class)->createAndRender($name, $props);
+        } catch (\Throwable $e) {
+            $this->throwRuntimeError($name, $e);
+        }
     }
 
     public function embeddedContext(string $name, array $props, array $context): array
     {
-        return $this->container->get(ComponentRenderer::class)->embeddedContext($name, $props, $context);
+        try {
+            return $this->container->get(ComponentRenderer::class)->embeddedContext($name, $props, $context);
+        } catch (\Throwable $e) {
+            $this->throwRuntimeError($name, $e);
+        }
+    }
+
+    private function throwRuntimeError(string $name, \Throwable $e): void
+    {
+        if (!($e instanceof \Exception)) {
+            $e = new \Exception($e->getMessage(), $e->getCode(), $e->getPrevious());
+        }
+        throw new RuntimeError(sprintf('Error rendering "%s" component: %s', $name, $e->getMessage()), previous: $e);
```

Comment:
I snuck this feature in. We may need some trial and error. But throwing a `RuntimeError` allows Twig to show the actual template + line number where the error occurred. The nested exception will be the one from your component (e.g. maybe you are passing `null` to a property in your component that only accepts `string`).